src/Interpreters/ProcessList.cpp

```diff
@@ -428,17 +434,22 @@ void QueryStatus::addPipelineExecutor(PipelineExecutor * e)
         throw Exception(ErrorCodes::QUERY_WAS_CANCELLED, "Query was cancelled");
 
     std::lock_guard lock(executors_mutex);
-    assert(std::find_if(executors.begin(), executors.end(), [e](const ExecutorHolderPtr & x){ return x->executor == e; }) == executors.end());
+    assert(std::find_if(executors.begin(), executors.end(), [e](ExecutorHolderPtr & x){ return x->equals(e); }) == executors.end());
```

Comment:
It will lock `ExecutorHolder::mutex` while holding `executors_mutex`, so here we are again

Reply:
Yeh, I already realozed it. I am thinking how to avoid it

Reply:
We can do same as in `cancelQuery`, first copy `executors` vector under mutex and then perform operations on `ExecutorHolder`

Reply:
I have a stupid idea: we can rename `equals` to something like `inprecieseDebugCheckEquals`, do `try_lock` there, and return `true` if `try_lock` failed. It's okay for a debug assertion